### ISSUE TITLE
Provide a better error handling with respect to ARIA functionality when showing error in Registry page

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop#5714]
### Is your enhancement related to a problem? Please describe

There is missing ARIA functionality for errors or warning showed in registry page.

### Describe the solution you'd like

Add missing aria functionality.

**Comment by trmendes** [comment]
Hi @odockal.

Just to be sure :).

When you say registry page. Are you talking about the page to type credentials, login and, enable access to some external hubs or some other page?

**Comment by odockal** [comment]
> Hi [@odockal](https://github.com/odockal).
> 
> Just to be sure :).
> 
> When you say registry page. Are you talking about the page to type credentials, login and, enable access to some external hubs or some other page?

yes (sorry for late reply). But these efforts got new wind from our UX designers, not sure if we will track it via issue.

**Comment by odockal** [comment]
@cbr7 I would rather have some input from @vancura @MariaLeonova how to address these parts in general. Let's keep it out of this sprint.

**Comment by vancura** [comment]
@odockal: Is there any way to test how errors and/or warnings look there? What can I break to see them? Then I can say what we need a11y-wise :)

**Comment by odockal** [comment]
@vancura at least these two. 
* Adding invalid registry (`https://registy.io` + some username and some pass)
* Add existing registry but invalid username/pass

**Comment by vancura** [comment]
I hope I understand correctly that we are talking about this area:

<img width="1208" height="838" alt="Image" src="https://github.com/user-attachments/assets/54d35473-0aff-4f08-a0ef-ee6ea7340123" />

If so, there are a couple of improvements we can make:

For the icon-only mode (https://github.com/podman-desktop/podman-desktop/blob/main/packages/ui/src/lib/alert/ErrorMessage.svelte#L33-L37):

1. There's a missing semantic information: the component only shows a tooltip with the error icon, but it provides no `role="alert"` or other ARIA labls for screen readers.
2. The error message is available only via a tooltip after a hover, which is not accessible to keyboard / screen reader / touch device users. Tooltips are generally horrible in regard to a11y. We should display the message without requiring a hover.
3. Screen readers will not automatically read the error when it appears.
4. The icon has `cursor-pointer` styling but no `tabindex`, making it inaccessible to keyboard users who can't hover.

For the full message mode (https://github.com/podman-desktop/podman-desktop/blob/main/packages/ui/src/lib/alert/ErrorMessage.svelte#L39-L44):

1. The icon there lacks semantic meaning. It has no `aria-labl` or a `role`, which makes it meaningless to screen readers.
2. The `aria-label` on the `div` hides the real message. Screen readers would announce “Error Message Content” instead of the error text.
3. Keeping a `role="alert"` in the tree while visually hidden by CSS (`opacity-0`) can lead to empty alerts and unreliable announcements. `role="alert"` should appear (or its text should change) when the error occurs.

Here's what we can do for the icon-only mode:

- Keep a screen-reader-only alert in the DOM when there’s an error so it is announced, even if we show only an icon AND tooltip.
- Make the icon focusable (make it a button) and associate it with the hidden alert via `aria-describedby`.
- Keep the visual tooltip purely visual (`aria-hidden=true`) to avoid duplicate announcements.

And for the full-message mode:

- Remove `aria-label` from the `div` element. Let the text be the accessible name.
- Only render the alert when there is a non-empty error string, so insertion/change reliably triggers the announcement.
- Optionally expose `id` so inputs can reference the error via `aria-errormessage` or `aria-describedby`.

For the `Tooltip` component (https://github.com/podman-desktop/podman-desktop/blob/main/packages/ui/src/lib/tooltip/Tooltip.svelte):

1. Use `role="tooltip"` on the tooltip panel (not `aria-label="tooltip"`).
2. Support focus as a trigger (e.g., `.tooltip-slot:focus-within + .tooltip`). The trigger should be focusable in the icon-only path.
3. Optionally give the tooltip an `id` and set `aria-describedby` on the trigger, but if we keep a separate screen-reader-only alert, we should mark the tooltip `aria-hidden="true"` so it doesn’t read both.

  This would be a good improvement and would help the component follow WCAG 2.1 guidelines, particularly [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) and [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html). It is particularly important for error messages since they're critical user feedback.

**Comment by odockal** [comment]
@vancura nice. I personally have no strong preference between icon mode and full-message mode, I can imagine some kind of hybrid where the main error is visible as it comes, but if there is more info/stack to be shown, then it should be hidden and have option to be shown.

I am ok with anything that is easily testable for error is there and the error message is `aaa` or `bbb`.

In general, this is just one area. We are using the same patterns through the app, so unified approach to be applied to other errors based on the context where they are shown.
